src/impl/vamp/random/halton.hh

```diff
@@ -25,42 +28,58 @@ namespace vamp::rng
             53.F,
             59.F};
 
-        explicit Halton(FloatVector<dim> b_in, std::size_t skip_iterations = 0) noexcept : b(b_in)
+        explicit Halton(FloatVector<dim> b_in) noexcept : b_init(b_in), b(b_in)
         {
-            initialize(skip_iterations);
         }
 
-        Halton(std::initializer_list<FloatT> v, std::size_t skip_iterations = 0) noexcept
-          : Halton(FloatVector<dim>::pack_and_pad(v), skip_iterations)
+        Halton(std::initializer_list<FloatT> v) noexcept : Halton(FloatVector<dim>::pack_and_pad(v))
         {
         }
 
-        explicit Halton(std::size_t skip_iterations = 0)
-          : Halton(
-                []()
-                {
-                    alignas(FloatVectorAlignment) std::array<float, dim> a;
-                    std::copy_n(primes.cbegin(), dim, a.begin());
-                    return a;
-                }(),
-                skip_iterations)
+        explicit Halton() : Halton(bases())
         {
         }
 
-        inline auto initialize(std::size_t n) noexcept
+        inline constexpr auto bases() noexcept -> FloatVector<dim>
         {
-            for (auto i = 0U; i < n; ++i)
-            {
-                next();
-            }
+            alignas(FloatVectorAlignment) std::array<float, dim> a;
+            std::copy_n(primes.cbegin(), dim, a.begin());
+            return FloatVector<dim>(a);
+        }
+
+        auto rotate_bases() noexcept
+        {
+            alignas(FloatVectorAlignment) std::array<float, dim> a;
+            b.to_array(a.data());
+            std::rotate(a.begin(), a.begin() + 1, a.end());
+            b = FloatVector<dim>(a);
         }
 
+        std::size_t iterations = 0;
+        FloatVector<dim> b_init;
```

Comment:
Should probably make this `const`. It might hurt the `move`ability of this type, but I don't think we're passing these by value anywhere/benefiting from move semantics anyway.